### PR TITLE
Move from Slack to Datadog, use crashloopbackon label

### PR DESCRIPTION
**Pull request by pcarrier**
This migrates notifications from Slack posts to Datadog events.
It also limits the scope of the tool to pods labeled with `crashloopbackon` = `true`.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -45,39 +42,54 @@ func main() {
...
-									log.Printf("Rate limited, sleeping %s", rlError.RetryAfter)
-									time.Sleep(rlError.RetryAfter)
-									break
+		if v, found := pod.Labels["crashloopbackon"]; found {
+			if b, _ := strconv.ParseBool(v); !b {
+				continue watchloop
+			}
+			switch evt.Type {
+			case "ADDED", "MODIFIED":
+				if v := pod.Labels["crashloopbackon"]; v == "true" {
```
Comment (glasser) [inline main.go]: Didn't you already check this above?

File: main.go [hunk 2]
```diff
@@ -45,39 +42,54 @@ func main() {
...
-						log.Printf("Pod %v/%v is in CrashLoopBackOff, deleting", pod.Namespace, pod.Name)
-						for {
-							if _, _, err = chat.PostMessage(*channelId, slack.MsgOptionText(
-								fmt.Sprintf("Deleting crashlooping pod `%s/%s`", pod.Namespace, pod.Name), false)); err != nil {
-								var rlError *slack.RateLimitedError
-								if errors.As(err, &rlError) {
-									log.Printf("Rate limited, sleeping %s", rlError.RetryAfter)
-									time.Sleep(rlError.RetryAfter)
-									break
+		if v, found := pod.Labels["crashloopbackon"]; found {
```
Comment (glasser) [inline main.go:46]: Wasn't there supposed to be a com.apollographql/?

File: main.go [hunk 3]
```diff
@@ -45,39 +42,54 @@ func main() {
...
+func handle(pod *v1.Pod, cs *kubernetes.Clientset, ctx context.Context, events *datadogV1.EventsApi, containerName string) error {
+	tags := []string{
+		fmt.Sprintf("namespace:%s", pod.Namespace),
+		fmt.Sprintf("pod_name:%s", pod.Name),
+		fmt.Sprintf("workload:%s", containerName),
+	}
+	if pteam, found := pod.Labels["com.apollographql/primaryTeam"]; found {
+		tags = append(tags, fmt.Sprintf("team:%s", pteam))
+	}
+	if _, _, err := events.CreateEvent(ctx, datadogV1.EventCreateRequest{
```
Comment (glasser) [inline main.go:76]: Not sure I remember the implication of DD events or how they'll show up (or why better than Slack directly)

File: main.go (hunk 2 above)
Comment (pcarrier) [reply main.go:46]: It felt unnecessary.

File: main.go (hunk 3 above)
Comment (pcarrier) [reply main.go:76]: Dropping the move to DD.

**Pull request closed without merging**